src/physics.cpp

```diff
@@ -140,8 +140,26 @@ void sample_neutron_reaction(Particle& p)
 
   // Sample a scattering reaction and determine the secondary energy of the
   // exiting neutron
-  scatter(p, i_nuclide);
 
+#ifdef NCRYSTAL
+  if (model::materials[p.material()]->m_NCrystal_mat() != nullptr && p.E() < settings::ncrystal_max_energy){
+    NcrystalRNG_Wrapper rng(p.current_seed()); // Initialize RNG
+    //create a cache pointer for multi thread physics
+    NCrystal::CachePtr dummyCache;//fixme: avoid recreating here (triggers malloc)
```

Comment:
Concerning this fixme: The code here works, but will incur a CPU overhead which is not strictly needed. Ideally each material and each thread would have its own `NCrystal::CachePtr` object, that would be kept around and passed to the `sampleScatter(..)` call in this function. If we could be 100% sure to not have to care about multi-thread safety, we could simply add an `NCrystal::CachePtr` object on the openmc `Material` object next to the shared pointer for the NCrystal process that we have there now (i.e. inside the `ifdef NCRYSTAL` section of material.h).

We left the code as is for now, since it is certainly safe and will give correct results (it will merely make NCrystal look slower than it actually is) and it seemed more beneficial to try to move ahead with the NCrystal+OpenMC integration since the issue is almost certainly ortogonal to any other potential issues concering this integration.

When openmc experts have time to discuss with us how they would like per-material per-thread caches kept (thread-local storage on `Material`?), we can create a new PR addressing this.

Reply:
@tkittel I think this is the right approach for now. Let's stick with the slow but safe approach -- once this is merged, we can create an issue to discuss possible implementations of a per-material, per-thread cache.